### ISSUE TITLE
Backup strategy for timeentries and other critical data

### ISSUE THREAD

**Issue by okms** [Puzzlepart/did#245]
### TLDR ###
In the case of corruption of the TimeEntries table and other tables, it should be possible to do a rollback and restore to a working version

### Discussion ###
The data in the TimeEntries table is accumulated as users confirm their weeks. In the case of data loss or corruption there currently is no way to iterate all calendars programatically and re-build the TimeEntries as they were. The fact that users can manually correct and confirm calendar events means re-building TimeEntries cannot even be done with 100% accuracy. As such the TimeEntries table is critical data for any organization using did and should be regularly backed up. 

This issue is to discuss possible backup strategies and how to implement them in Did 365. 

Currently i can think of the following options

1) backup from one storage account to another using [Azcopy](https://docs.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-v10).

Can be done using Azure automation or from the command line on a PC. Seems this is the most common option

2) adding a "backup" function to the admin interface. ideally this interface should allow us to set up a periodic bacup and also a location to store file dumps of backups. This would be a more or less 100% custom solution 

3) .. your suggestions here
